install-wings.sh

```diff
@@ -447,68 +394,83 @@ function firewall_ufw {
   ufw status numbered | sed '/v6/d'
 }
 
-function firewall_firewalld {
+firewall_firewalld() {
   echo -e "\n* Enabling firewall_cmd (firewalld)"
   echo "* Opening port 22 (SSH), 8080 (Daemon Port), 2022 (Daemon SFTP Port)"
 
   # Install
-  [ "$OS_VER_MAJOR" == "7" ] && yum -y -q update
   [ "$OS_VER_MAJOR" == "7" ] && yum -y -q install firewalld > /dev/null
-  [ "$OS_VER_MAJOR" == "8" ] && dnf -y -q update
   [ "$OS_VER_MAJOR" == "8" ] && dnf -y -q install firewalld > /dev/null
 
   # Enable
   systemctl --now enable firewalld > /dev/null # Enable and start
 
   # Configure
+  firewall-cmd --ad-service=ssh --permanent -q # Port 22
   firewall-cmd --add-port 8080/tcp --permanent -q # Port 8080
   firewall-cmd --add-port 2022/tcp --permanent -q # Port 2022
-  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-port 80/tcp --permanent -q # Port 80
-  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-port 443/tcp --permanent -q # Port 443
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-service=http --permanent -q # Port 80
+  [ "$CONFIGURE_LETSENCRYPT" == true ] && firewall-cmd --add-service=https --permanent -q # Port 443
 
   firewall-cmd --permanent --zone=trusted --change-interface=pterodactyl0 -q
   firewall-cmd --zone=trusted --add-masquerade --permanent
-  firewall-cmd --ad-service=ssh --permanent -q # Port 22
   firewall-cmd --reload -q # Enable firewall
 
   echo "* Firewall-cmd installed"
   print_brake 70
 }
 
+letsencrypt() {
+  FAILED=false
+
+  # Install certbot
+  if [ "$OS" == "debian" ] || [ "$OS" == "ubuntu" ]; then
+    apt-get install certbot -y
+  elif [ "$OS" == "centos" ]; then
+    [ "$OS_VER_MAJOR" == "7" ] && yum install certbot
+    [ "$OS_VER_MAJOR" == "8" ] && dnf install certbot
+  else
+    # exit
+    print_error "OS not supported."
+    exit 1
+  fi
+
+  # If user has nginx
+  systemctl stop nginx || true
+
+  # Obtain certificate
+  certbot certonly --no-eff-email --email "$EMAIL" --standalone -d "$FQDN" || FAILED=true
+
+  systemctl start nginx || true
+
+  # Check if it succeded
+  if [ ! -d "/etc/letsencrypt/live/$FQDN/" ] || [ "$FAILED" == true ]; then
+    print_warning "The process of obtaining a Let's Encrypt certificate failed!"
+  fi
+}
+
 ####################
 ## MAIN FUNCTIONS ##
 ####################
-function perform_install {
+
+perform_install() {
   echo "* Installing pterodactyl wings.."
-  [ "$CONFIGURE_UFW" == true ] && firewall_ufw
-  [ "$CONFIGURE_FIREWALL_CMD" == true ] && firewall_firewalld
-  install_dep
+  [ "$OS" == "ubuntu" ] || [ "$OS" == "debian" ] && apt_update
+  [ "$OS" == "centos" ] || [ "$OS_VER_MAJOR" == "7" ] && yum_update
+  [ "$OS" == "centos" ] || [ "$OS_VER_MAJOR" == "8" ] && dnf_update
+  "$CONFIGURE_UFW" && firewall_ufw
+  "$CONFIGURE_FIREWALL_CMD" && firewall_firewalld
```

Comment:
Nice refactor, a lot cleaner than checking for the value specifically! :+1:

Reply:
Thanks :+1: